parent: !include default.yaml
general_options:
  joint_vel_limit: # giskard will use the min of this number and limits from the urdf
    linear:
      default: 0.5
      odom_x_joint: 0.5
      odom_y_joint: 0.5
    angular:
      default: 0.5
      odom_z_joint: 0.5
  joint_weights: # the planner prefers low weight joints to achieve solutions
    default: 0.01
    odom_x_joint: 0.05
    odom_y_joint: 0.05
    odom_z_joint: 0.05
plugins:
  PlotTrajectory: # plots the joint trajectory at the end of planning
    enabled: False
    velocity_threshold: 0.0 # only joints that exceed this velocity threshold will be added to the plot. Use a negative number if you want to include every joint
    scaling: 5.0 # determines how much the x axis is scaled with the length(time) of the trajectory
    normalize_position: False # centers the joint positions around 0 on the y axis
    tick_stride: 1.0 # the distance between ticks in the plot. if tick_stride <= 0 pyplot determines the ticks automatically 
  WiggleCancel: # cancels planning if high frequencies are detected in the trajectory
    amplitude_threshold: 0.15 # the amplitude of a frequency must exceed this threshold to be detected as wiggling
    window_size: 21 # size of the moving window of sample points
    frequency_range: 0.4 # must be in the range [0,1]. Only frequencies in the range [max_detectable_frequency * wiggle_frequency_range, max_detectable_frequency] are considered as wiggling. So a value of 0 means that every frequency with an amplitude above wiggle_detection_threshold is detected as wiggling and a value of 1 means that only the max_detectable_frequency can trigger a wiggle detection.      max_detectable_frequency = 1 / (2 * sample_period)
reachability_check:
  sample_period: 0.5
  prismatic_velocity: 2.0
  continuous_velocity: 0.5
  revolute_velocity: 0.5
  other_velocity: 0.5
